swan-lake/development-tutorials/observability/supported-observability-tools-and-platforms/opensearch.md

```diff
@@ -0,0 +1,747 @@
+
+    networks:
+      observability:
+        driver: bridge
+   ```
+
+3. Create the following configuration files in the `config` folder.
+
+   - `config/ballerina/Config.toml`
+
+   Use the same content as in the previous step, which enables metrics logs and tracing in the Ballerina application and configures the Jaeger exporter.
+   
+   - `config/dashboards/pipelines.yaml`
+    
+   ```yaml
+    # Server configuration
+    server.name: opensearch-dashboards
+    server.host: "0.0.0.0"
+    server.port: 5601
+      
+    # OpenSearch connection
+    opensearch.hosts: ["http://opensearch:9200"]
+      
+    opensearch.ssl.verificationMode: none
+    opensearch.username: "admin"
+    opensearch.password: "Ballerina@123"
+    opensearch.requestHeadersWhitelist: [authorization, securitytenant]
+
+    opensearch_security.multitenancy.enabled: true
+    opensearch_security.multitenancy.tenants.preferred: [Private, Global]
+    opensearch_security.readonly_mode.roles: [kibana_read_only]
+    # Use this setting if you are running opensearch-dashboards without https
+    opensearch_security.cookie.secure: false
+
+    # Branding configuration
+    opensearchDashboards.branding.logo.defaultUrl: "https://raw.githubusercontent.com/wso2/observability-resources/main/images/wso2-dashboard-logo.png"
+    opensearchDashboards.branding.mark.defaultUrl: "https://raw.githubusercontent.com/wso2/observability-resources/main/images/wso2-dashboard-icon.png"
+    opensearchDashboards.branding.loadingLogo.defaultUrl: "https://raw.githubusercontent.com/wso2/observability-resources/main/images/loading.png"
+    opensearchDashboards.branding.applicationTitle: "WSO2 Dashboard"
+
+    # Logging configuration for debugging
+    logging.dest: stdout
+    logging.silent: false
+    logging.quiet: false
+   ```
+   
+   - `config/data-prepper/pipelines.yaml`
+    
+   ```yaml
+    entry-pipeline:
+      delay: "100"
+      source:
+        otel_trace_source:
+          port: 4317
+          ssl: false
+      sink:
+        - pipeline:
+            name: "raw-pipeline"
+        - pipeline:
+            name: "service-map-pipeline"
+        - stdout: {}
+   
+    raw-pipeline:
+      source:
+        pipeline:
+          name: "entry-pipeline"
+      processor:
+        - otel_traces:
+      sink:
+        - opensearch:
+            hosts: [ "https://opensearch:9200" ]
+            username: "admin"
+            password: "Ballerina@123"
+            insecure: true        
+            index_type: trace-analytics-raw
+    
+    service-map-pipeline:
+      delay: "100"
+      source:
+        pipeline:
+          name: "entry-pipeline"
+      processor:
+        - service_map:
+      sink:
+        - opensearch:
+            hosts: [ "https://opensearch:9200" ]
+            username: "admin"
+            password: "Ballerina@123"
+            insecure: true
+            index_type: trace-analytics-service-map
+   ```
+
+   - `config/fluent-bit/fluent-bit.conf`
+
+   ```conf
```

Comment:
`conf` is not supported by shiki. See whether there is an alternate language code we can use for this.

Reply:
added apache as the language code